src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -6,8 +6,32 @@
 
 namespace Microsoft.Extensions.Primitives
 {
+    /// <summary>
+    /// An optimized representation of a substring.
+    /// </summary>
     public struct StringSegment : IEquatable<StringSegment>, IEquatable<string>
     {
+        /// <summary>
+        /// Initializes an instance of the <see cref="StringSegment"/> struct.
+        /// </summary>
+        /// <param name="buffer">
+        /// The <see cref="string"/>that acts as a buffer for this <see cref="StringSegment"/>.
+        /// In this case, the <see cref="StringSegment"/> represents the whole original <see cref="string"/>
+        /// or its absence in case the parameter is <code>null</code>.
```

Comment:
Can you provide a suggestion? I want to remark that this constructor is for creating a StringSegment from the given string and that null buffer is the "equivalent" of null string in this case.